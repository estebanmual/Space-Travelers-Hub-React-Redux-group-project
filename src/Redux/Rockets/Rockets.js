import { FETCH_ROCKETS } from '../../constants';

export const fetchRockets = () => async (dispatch) => {
  const response = await fetch('https://api.spacexdata.com/v3/rockets');
  const data = await response.json();
  dispatch({ type: FETCH_ROCKETS, payload: data });
};

export const rocketsReducer = (state = [], action) => {
  const rockets = [];
  switch (action.type) {
    case FETCH_ROCKETS:
      action.payload.forEach((rocket) => {
        rockets.push({
          id: rocket.rocket_id,
          name: rocket.rocket_name,
          type: rocket.rocket_type,
          flickrImages: rocket.flickr_images[0],
        });
      });
      return rockets;
    default:
      return state;
  }
};
