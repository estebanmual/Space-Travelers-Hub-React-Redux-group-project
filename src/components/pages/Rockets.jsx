import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../../Redux/Rockets/Rockets';

function Rockets() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  console.log(rockets);

  useEffect(() => {
    dispatch(fetchRockets());
  }, []);
  return <div>Rockets</div>;
}

export default Rockets;
