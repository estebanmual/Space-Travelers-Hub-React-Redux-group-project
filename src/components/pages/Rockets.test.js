import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../Redux/storeSetUp';
import Rockets from './Rockets';

describe('testing Header', () => {
  it('snapshot for the header ', () => {
    const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Header component renders correctly', () => {
    render(<Provider store={store}><Rockets /></Provider>);
    expect(screen.getByTestId('rockets')).toBeInTheDocument();
  });
});
