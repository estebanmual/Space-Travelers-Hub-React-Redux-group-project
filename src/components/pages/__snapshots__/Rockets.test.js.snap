// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header snapshot for the header  1`] = `null`;
